conan/tools/gnu/pkgconfigdeps.py

```diff
@@ -48,33 +48,35 @@ class PkgConfigDeps(object):
     def __init__(self, conanfile):
         self._conanfile = conanfile
 
-    def _get_require_comp_name(self, pkg_name, req):
+    def _get_require_comp_name(self, dep, req):
+        pkg_name = dep.ref.name
         pkg, comp_name = req.split("::") if "::" in req else (pkg_name, req)
-        req = self._conanfile.dependencies.direct_host[pkg]
+        req = dep.dependencies.direct_host[pkg]
```

Comment:
@memsharded this line is working fine for all the cases but for components depending on another component. For instance:

```python
        self.cpp_info.components["myothercomp"].requires.append("mycomponent")
```
This one is failing due to it's not able to find out the given component ref, however, if you change this line by:

```python
       req = self._conanfile.dependencies.host[pkg]
```
All the tests are passing without any problem.

Reply:
The problem with ``req = self._conanfile.dependencies.host[pkg]`` is that it could allow defining ``requires`` to not direct dependencies which is bad. But probably this responsibility shouldn't be in PkgConfigDeps, so I am fine with this fix. Please make the tests green!